src/lib/storage/lz4_segment.cpp

```diff
@@ -77,8 +77,14 @@ std::optional<T> LZ4Segment<T>::get_typed_value(const ChunkOffset chunk_offset)
 }
 
 template <typename T>
-const std::optional<pmr_vector<bool>>& LZ4Segment<T>::null_values() const {
-  return _null_values;
+bool LZ4Segment<T>::contains_nulls() const {
+  return _null_values.has_value();
+}
+
+template <typename T>
+const pmr_vector<bool>& LZ4Segment<T>::null_values() const {
+  DebugAssert(contains_nulls(), "This LZ4Segment does not contain a NULL vector.");
```

Comment:
As the problem is usually printed to the command line, reading "this segment" is strange because I don't get any information what "this is". Maybe just "LZ4Segment does not ..." ?